torch/csrc/jit/script/compiler.cpp

```diff
@@ -907,6 +1104,27 @@ struct to_ir {
     return v;
   }
 
+  // if a value is only set by an unchecked unwrap optional, than the value
+  // has not changed and we omit adding an if node output for it.
+  // this also prevents jitter when importing an if expression like:
+  // x = x if x is not None else 1
+  bool valueNotWrittenTo(
```

Comment:
I think this is best handled in dead code elimination rather than here, where it makes the compiler logic more complicated. Dead code elimination already deletes unneeded if statement/while outputs. It is pretty natural to add a rule there to make it understand that an if output can be eliminated if it is simply combining an unchecked_unwrap(x) with an x or another unchecked_unwrap(x).

Also, I believe this might be buggy. Just because an output of an is is an _uncheck_unwrap_optional does not me it is the  _same variable_ as it originally was. There doesn't appear to be any check that true_v's input is the original value for v. For instance, what happens if someone swaps two unwrapped optionals in an if statement.